api/python/quilt3/packages.py

```diff
@@ -910,11 +915,20 @@ def set_dir(self, lkey, path=None, meta=None, update_policy="incoming"):
             src_path = src.path
             if src.basename() != '':
                 src_path += '/'
-            objects, _ = list_object_versions(src.bucket, src_path)
+            try:
+                objects, _ = list_object_versions(src.bucket, src_path)
+            except botocore.exceptions.ClientError as e:
+                if not allow_unversioned or e.response["Error"]["Code"] != "AccessDenied":
+                    raise e
+                print(f"Warning[s3://{src.bucket}/{src_path}]: {e}.")
+                objects = list_objects(src.bucket, src_path, recursive=True)
+                for obj in objects:
+                    obj["IsLatest"] = True
+
             for obj in objects:
                 if not obj['IsLatest']:
                     continue
```

Comment:
I don't understand.   I am not filtering on the objects WITH VersionId.  I am *adding* a pseudo-version to the objects *without* VersionId, so that I don't have to change the downstream code.

Reply:
Ah.  So you are changing the downstream code.  VS Code mangled the suggestion, but I'll try to implement that.